analyzers/src/SonarAnalyzer.Common/SymbolicExecution/Roslyn/OperationProcessors/Invocation.cs

```diff
@@ -51,6 +51,10 @@ protected override ProgramState[] Process(SymbolicContext context, IInvocationOp
         {
             state = state.ResetFieldConstraints();
         }
+        if (invocation.TargetMethod.IsStatic)
```

Comment:
I'm not sure, how we should handle extension methods. We already do check that the `ContainingType is the same, so only calls from one extension method to another extension method of the same class will be ever processed and there we would reset static fields stored in the extension method class (whoever does such things?).
However, the current behavior is not documented by tests by now.

Reply:
Extension method is a very specific case that should not touch the current class, so we're fine. I'd leave it as it is.